diff --git a/sac_continuous_action.py b/sac_continuous_action.py
index f494e51..583e573 100644
--- a/sac_continuous_action.py
+++ b/sac_continuous_action.py
@@ -27,9 +27,9 @@ class Args:
     """if toggled, `torch.backends.cudnn.deterministic=False`"""
     cuda: bool = True
     """if toggled, cuda will be enabled by default"""
-    track: bool = False
+    track: bool = True
     """if toggled, this experiment will be tracked with Weights and Biases"""
-    wandb_project_name: str = ""
+    wandb_project_name: str = "Quad_Mujoco"
     """the wandb's project name"""
     wandb_entity: Optional[str] = None
     """the entity (team) of wandb's project"""
